src/main/java/uz/pdp/eufloria/service/impl/StoreServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package uz.pdp.eufloria.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uz.pdp.eufloria.domain.Store;
+import uz.pdp.eufloria.dto.StoreDto;
+import uz.pdp.eufloria.exception.AlreadyExistsException;
+import uz.pdp.eufloria.exception.NotFoundException;
+import uz.pdp.eufloria.exception.NullOrEmptyException;
+import uz.pdp.eufloria.repository.StoreRepository;
+import uz.pdp.eufloria.service.StoreService;
+import uz.pdp.eufloria.util.Validator;
+
+import java.time.LocalTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class StoreServiceImpl implements StoreService {
+    private final StoreRepository storeRepository;
+
+    @Override
+    public StoreDto save(StoreDto storeDto) {
+        if (storeDto == null)
+            throw new NotFoundException("Store");
+        if (Validator.isNullOrEmpty(storeDto.getName()))
+            throw new NullOrEmptyException("Store name");
+        if (storeDto.getOpens() == null)
+            throw new NullOrEmptyException("Store opens");
+        if (storeDto.getCloses() == null)
+            throw new NullOrEmptyException("Store closes");
+        if (storeDto.getId() != null) {
+            if (storeRepository.findById(storeDto.getId()).isPresent())
+                throw new AlreadyExistsException("Store id");
+        }
+        if (storeRepository.findByName(storeDto.getName()).isPresent())
+            throw new AlreadyExistsException("Store name");
+
```

Comment:
Use validator annotations rather if else checking

Reply:
It's done